cnxpublishing/views/admin.py

```diff
@@ -271,3 +274,86 @@ def admin_edit_site_message_POST(request):
     args = admin_edit_site_message(request)
     args['response'] = "Message successfully Updated"
     return args
+
+
+@view_config(route_name='admin-print-style', request_method='GET',
+             renderer='cnxpublishing.views:templates/print-style.html',
+             permission='administer')
+def admin_print_styles(request):
+    settings = request.registry.settings
+    db_conn_str = settings[config.CONNECTION_STRING]
+    styles = []
+    with psycopg2.connect(db_conn_str) as db_conn:
+        with db_conn.cursor() as cursor:
+            cursor.execute("""\
+                SELECT print_style, fileid, recipe_type, revised, tag,
+                    (SELECT count (*) from latest_modules as lm
+                        where lm.print_style=ps.print_style
+                            and lm.portal_type='Collection')
+                FROM print_style_recipes as ps;""")
+            for row in cursor.fetchall():
+                styles.append({
+                    'print_style': row[0],
+                    'file': row[1],
+                    'type': row[2],
+                    'revised': row[3],
+                    'tag': row[4],
+                    'number': row[5],
+                    'link': request.route_path('admin-print-style-single',
+                                               style=row[0])
+                })
+    return {'styles': styles}
+
+
+@view_config(route_name='admin-print-style-single', request_method='GET',
+             renderer='cnxpublishing.views:templates/print-style-single.html',
+             permission='administer')
+def admin_print_styles_single(request):
+    settings = request.registry.settings
+    db_conn_str = settings[config.CONNECTION_STRING]
+    style = request.matchdict['style']
+    args = {}
+    # do db search to get file id and other info on the print_style
+    with psycopg2.connect(db_conn_str) as db_conn:
+        with db_conn.cursor() as cursor:
+            cursor.execute("""
+                SELECT print_style, fileid, recipe_type, tag
+                from print_style_recipes
+                WHERE print_style=%s
+                """, vars=(style,))
+            info = cursor.fetchall()
+            if len(info) != 1:
```

Comment:
I see now. The version of the database I had had print style as unique, but the db has changed since. I think it should show books with all the versions of the print style, so it will get any book with that print style and then also display the recipe and tag in the table.